src/main/java/xyz/bluspring/kilt/KiltMixinPlugin.java

```diff
@@ -34,7 +34,7 @@ public void onLoad(String mixinPackage) {
 
         ExtensionRegistrar.register(new KiltMixinModifier());
 
-        Kilt.Companion.getLoader().scanMods();
+        Kilt.Companion.getLoader().getScanModJob();
```

Comment:
What does this do, actually? It doesn't appear to be doing much, so I'm not sure what this is actually meant to do.

Reply:
The scan mods is executing off thread here.
When the method called on Main thread. The mixin processor is locked since the call site class is a mixin plugin.
Then, a deadlock will appear when transform the classes.
I have to execute it off thread to avoid the deadlock.

Maybe I pick the wrong point to sync the method result to the main thread

Reply:
We need it complete before `KiltEarlyRiser#run`. Right?

Reply:
Correct.

Reply:
Why the remapper before won't run into it hmm

Reply:
found the reason :)

Reply:
It still seems to completely lock up on startup on the MixinProcessor, which makes me wonder if the mod remapping is simply not feasible to do in the mixin plugin.

Reply:
The only way is using java and treat the classes to be used carefully

Reply:
hmm.. that's a problem.
Let me see what I can do.